ledger/sandbox-on-x/src/classic/scala/com/daml/platform/ledger/SandboxServer.scala

```diff
@@ -104,25 +118,94 @@ final class SandboxServer(
       .map(path => Future(Files.write(path, Seq(port.toString).asJava)).map(_ => ()))
       .getOrElse(Future.unit)
 
-  private def loadPackages(writeService: WriteService)(implicit
-      executionContext: ExecutionContext
-  ): AbstractResourceOwner[ResourceContext, List[Unit]] =
-    ResourceOwner.forFuture(() =>
-      Future.sequence(
-        config.damlPackages.map { file =>
-          val submissionId = Ref.SubmissionId.assertFromString(UUID.randomUUID().toString)
-          for {
-            dar <- Future.fromTry(DarParser.readArchiveFromFile(file).toTry)
-            _ <- writeService
-              .uploadPackages(submissionId, dar.all, None)(
-                LoggingContext.ForTesting,
-                NoOpTelemetryContext,
-              )
-              .asScala
-          } yield ()
+  private def loadPackages(writeService: WriteService, indexService: IndexService)(implicit
+      executionContext: ExecutionContext,
+      system: ActorSystem,
+      loggingContext: LoggingContext,
+  ): AbstractResourceOwner[ResourceContext, Unit] =
+    ResourceOwner.forFuture(() => {
+      val packageSubmissionsTrackerMap = config.damlPackages.map { file =>
+        val uploadCompletionPromise = Promise[Unit]()
+        UUID.randomUUID().toString -> (file, uploadCompletionPromise)
+      }.toMap
+
+      scheduleUploadTimeout(packageSubmissionsTrackerMap.iterator.map(_._2._2), 10.seconds)
```

Comment:
1. So the time starts ticking for all packages even before we kick-off their loading. Will that not be a problem? 
2. In the failed CI runs we timeout after 30s. Is there no risk that the tests will be more flaky with a 10s timeout?

Reply:
1. Not sure if it matter much when compared with a 10 second period. But it looks strange, I agree. Changed
2. 10 seconds is much as well imho, but since I don't have a good rule of thumb value, let's go with 30, as you suggested.

Reply:
100f3a58d4db674aa48b8a73d4c424b6c481c37b